emcc.py

```diff
@@ -2369,11 +2388,17 @@ def generate_html(target, options, js_target, target_basename,
   } else {
     // note: no support for code mods (PRECISE_F32==2)
     console.log('running code on the main thread');
+    var filename = '%s';
+    var fileBytes = tryParseAsDataURI(filename);
```

Comment:
all the invocations of `tryParseAsDataURI` could be emitted only when `SINGLE_FILE`, I think?